### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by matrix880**

### CODE REVIEW

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 1]
```diff
@@ -6,18 +6,42 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int [] principaldiagonal = new int [matrix.length];
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]:
```suggestion
        int[] principalDiagonal = new int[matrix.length];
```
see the [guide](https://google.github.io/styleguide/javaguide.html#s5.1-identifier-names) to make correct naming

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 2]
```diff
@@ -6,18 +6,42 @@
...
+        for (int i = 0; i < matrix.length; i++) {
+            principaldiagonal[i] = matrix[i][i];
+        }
+        return principaldiagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int [] secondarydiagonal = new int [matrix.length];
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]:
```suggestion
        int[] secondaryDiagonal = new int[matrix.length];
```

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 1 above)
Comment (matrix880) [reply src/main/java/core/basesyntax/MatrixOperation.java]: done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 2 above)
Comment (matrix880) [reply src/main/java/core/basesyntax/MatrixOperation.java]: done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 1 above)
Comment (matrix880) [reply src/main/java/core/basesyntax/MatrixOperation.java]: done

**Comment by Tarabarnik** [comment]
variables naming is not fixed

**Comment by matrix880** [comment]
> variables naming is not fixed

done

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 3]
```diff
@@ -5,19 +5,43 @@
  * максимального та мінімального елементів.
  */
 public class MatrixOperation {
-    public int[] getDiagonal(int[][] matrix) {
-        return null;
+    public int [] getDiagonal(int[][] matrix) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]:
```suggestion
    public int[] getDiagonal(int[][] matrix) {
```
why do you write this spaces?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 4]
```diff
@@ -5,19 +5,43 @@
...
+        int [] principalDiagonal = new int [matrix.length];
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]:
```suggestion
        int[] principalDiagonal = new int[matrix.length];
```
please refactor your code same to this suggestion

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 5]
```diff
@@ -5,19 +5,43 @@
...
+        for (int i = 0; i < matrix.length; i++) {
+            principalDiagonal[i] = matrix[i][i];
+        }
+        return principalDiagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int [] secondaryDiagonal = new int [matrix.length];
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]: and here

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 3 above)
Comment (matrix880) [reply src/main/java/core/basesyntax/MatrixOperation.java]: didnt notice.done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 4 above)
Comment (matrix880) [reply src/main/java/core/basesyntax/MatrixOperation.java]: done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 5 above)
Comment (matrix880) [reply src/main/java/core/basesyntax/MatrixOperation.java]: done

**Review by Tarabarnik — APPROVED** [review]

**Pull request closed without merging**